app/views/listings/show.html.erb

```diff
@@ -8,6 +8,5 @@
   <%= link_to "Back to see all tours", listings_path%> <br>
   <% if current_user.user_type == 'vendor' %>
     <%= link_to 'Edit Listing', edit_listing_path %> <br>
-    <%= link_to "Delete", listing_path(@listing), data: {turbo_method: :delete, turbo_confirm: "Are you sure?" }%>
```

Comment:
Because you added the dependencies in above, we should now still be able to have the delete button and it shouldn't cause an issue. Did you test it? (You'll need to add the dependent destroy on bookings also)